packages/permission-log-controller/src/PermissionLogController.ts

```diff
@@ -449,36 +385,35 @@ export class PermissionLogController extends BaseController<
    * @param request - The request object.
    * @returns The names of the requested permissions.
    */
-  // TODO: Replace `any` with type
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  getRequestedMethods(request: JsonRpcRequestWithOrigin<any>): string[] | null {
+  #getRequestedMethods(request: JsonRpcRequestWithOrigin): string[] | null {
+    const { params } = request;
     if (
-      !request.params ||
-      !request.params[0] ||
-      typeof request.params[0] !== 'object' ||
-      Array.isArray(request.params[0])
+      Array.isArray(params) &&
```

Comment:
Nice job reversing this. Do we need a check for `params` itself, to mimic the original logic?

``` suggestion
      params &&
      Array.isArray(params) &&
```